src/parser/mod.rs

```diff
@@ -2227,7 +2228,15 @@ impl<'a> Parser<'a> {
 
         Ok((
             field_defs,
-            self.expect_closing_angle_bracket(trailing_bracket)?,
+            if token == Token::Lt {
```

Comment:
There's a slight mismatch I'm thinking might not be worth having - the open delimeter token is configurable now but here the function still needs to figure out for itself what the value is for the closing token.
The function is a bit complicated on its own, specifically due to the `<` delimeter. then since duckdb syntax doesn't have that I'm thinking it makes sense to use a dedicated function that is simpler.

Something roughly like this I think should be all duckdb need, would it make sense to add this instead and have duckdb (and any future dialect sharing the same syntax) use it?
```rust
   fn parse_duck_db_struct_type_def<F>(
        &mut self,
    ) -> Result<Vec<StructField>, ParserError>
    {
        let start_token = self.peek_token();
        self.expect_keyword(Keyword::STRUCT)?;

        self.expect_token(Token::LParen)?;

        let field_defs = self.parse_comma_separated(Self::parse_struct_field_def)?;

        self.expect_token(Token::RParen)?;

        Ok(field_defs)
    }
```

Reply:
I have separate implementation at the beginning, but since they have quite many similarity so I merge them into once function.